5_descriptors/descriptor.py

```diff
@@ -0,0 +1,51 @@
+"""implement your own descriptor that logs when attrs of a class is modified"""
+import logging
+
+logging.basicConfig(level=logging.INFO)
+
+
+class AttributeLogger:
+
+    def __set_name__(self, owner,
+                     name):  # records field names. So, we can access more than one field.
+        self.public_name = name
+        self.private_name = '_' + name  # not sure why we need the protected private name
```

Comment:
if you don't use "_" in the name, you'll be updating the same value which leads to calling `__set__ ` function in recursion. just try it.